src/ocean/task/TaskPool.d

```diff
@@ -153,6 +153,36 @@ class TaskPool ( TaskT : Task ) : ObjectPool!(Task)
         return true;
     }
 
+    /***************************************************************************
+
+        Suspends the current task until all tasks in the pool have finished
+        executing.
+
+    ***************************************************************************/
+
+    public void awaitAllTasks ()
+    {
+        scope iter = this.new ReadOnlyBusyItemsIterator;
+        int count;
+
+        auto task = Task.getThis();
+        assert(task !is null, "Null task in TaskPool.await");
```

Comment:
I'd also replace this with sanity `enforce` as there are no other safeguards here.